javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -677,4 +683,91 @@ protected SymbolResolver getSymbolResolver() {
     // We need to expose it because we will need to use it to inject the SymbolSolver
     public static final DataKey<SymbolResolver> SYMBOL_RESOLVER_KEY = new DataKey<SymbolResolver>() {
     };
+
+
+    public enum TreeTraversal {
+        PREORDER, BREADTHFIRST, POSTORDER
+    }
+
+    public Iterator<Node> treeIterator(TreeTraversal traversal) {
+        switch (traversal) {
+            case BREADTHFIRST:
+                return new TreeVisitor.BreadthFirstIterator(this);
+            case POSTORDER:
+                return new TreeVisitor.PostOrderIterator(this);
+            case PREORDER:
+                return new TreeVisitor.PreOrderIterator(this);
+            default:
+                throw new IllegalArgumentException("Unknown traversal choice.");
+        }
+    }
+    
+    public Iterable<Node> treeIterable(TreeTraversal traversal) {
+        return () -> treeIterator(traversal);
```

Comment:
@ftomassetti an iterable is usable in a for-each loop, so I added this. Funky syntax!

Reply:
Wow!

Reply:
I Googled it :-D